src/features/mail-service/MailService.ts

```diff
@@ -18,15 +22,24 @@ export class MailService {
 
   async send(mailData: MailData) {
     const transporter = this.createTransporter();
+    const __dirname = path.resolve(); // This will get the project root folder
+    const filePath = path.join(__dirname, `./src/templates/${mailData.templateName}.html`); // Ensure templateName is provided in MailData
 
     try {
-      await transporter.sendMail({
+      const source = await fs.promises.readFile(filePath, 'utf-8'); // Read the HTML template from the file asynchronously
+      const template = handlebars.compile(source); // Compile the Handlebars template
```

Comment:
Create a function inside this MailService with name `getTemplate` and pass templateName as parameter. And use this code inside `getTemplate` function.

So, `getTemplate` template function works as below.

We send `templateName` to `getTemplate` function and this function should find the `${templateName}.html` file and compile with handerbars and return that compiled file.